### DISCUSSION TITLE
How to boot jetson Xavier NX from SSD

### DISCUSSION THREAD

**Discussion by NavSriv** [OE4T/meta-tegra#1419] [Q&A] [answered]
I would like to boot Jetson Xavier NX devkit from SSD. I have Samsung 500GB SSD mounted on the board. I did the following steps to build and flash the image for SSD (followed https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support) -

1. Downloaded https://github.com/OE4T/meta-tegra/tree/kirkstone-l4t-r32.7.x along with poky (kirkstone)
2. Added following lines in local.conf
   TNSPEC_BOOTDEV:jetson-xavier-nx-devkit-emmc = "nvme0n1p1"
   MACHINE ??= "jetson-xavier-nx-devkit-emmc"
3. bitbake and got xxx-image-jetson-xavier-nx-devkit-emmc-20231110185847.tegraflash.tar.gz
4. Untar tar.gz package
5. Connected OTG cable to host machine and jumper to put the board in recovery mode. Removed SD card from the board.
6. Executed ./initrd-flash
7. Once script is done. Removed Jumper and OTG cable and power off/on the board.
8. Nothing comes up now. Even on connecting SD with image, nothing appears on the HDMI.

What am I doing wrong here? Am I supposed to execute doflash too along with initrd-flash?

**Comment by dwalkes** [comment]
> Am I supposed to execute doflash too along with initrd-flash?

No, that shouldn't be necessary.

What do your flash logs look like?  Any problems noted? I'd try `sudo ./initrd-flash` especially if you noticed any errors during the flashing process.

> Even on connecting SD with image, nothing appears on the HDMI.

What do you see on the [serial console](https://jetsonhacks.com/2019/04/19/jetson-nano-serial-console/)?

Also you can try https://github.com/OE4T/tegra-demo-distro for a working turn-key setup.  I've verified with the same platform, branch MACHINE and NVMe boot as of https://github.com/OE4T/tegra-demo-distro/tree/a8bf34b60ce271a360d76c01209da73b600d3bda - see discussion at https://github.com/OE4T/meta-tegra/discussions/1403#discussioncomment-7541673

The last thing remaining if that doesn't work is your NVMe.  You could try following NVIDIA's jetpack instructions to flash to NVMe and verify the NVMe boots successfully in that scenario.

**Comment by NavSriv** [comment, marked as answer]
Executing "./doflash.sh --spi-only" along with "./initrd-flash" did the trick. I am able to boot from SSD now. Thanks for all the help!
